File: C++/Hard/CompileError/perfect-rectangle_8.cpp
class Solution {
public:
    bool isRectangleCover(vector<vector<int>>& rectangles) {
        
        int n = rectangles.size()(r[0]) + " " + to_string(r[3]);
            string p3 = to_string(r[2]) + " " + to_string(r[3]);
            string p4 = to_string(r[2]) + " " + to_string(r[1]);
            if (points.count(p1)) points.erase(p1);
            else points.insert(p1);
            if (points.count(p2)) points.erase(p2);
            else points.insert(p2);
            if (points.count(p3)) points.erase(p3);
            else points.insert(p3);
            if (points.count(p4)) points.erase(p4);
            else points.insert(p4);
        }
        if (!points.count(to_string(x1) + " " + to_string(y1)) ||
            !points.count(to_string(x1) + " " + to_string(y2)) ||
            !points.count(to_string(x2) + " " + to_string(y1)) ||
            !points.count(to_string(x2) + " " + to_string(y2)) ||
            points.size() != 4) return false;
        return area == (x2 - x1) * (y2 - y1);
    }
};

File: Java/Hard/Accepted/perfect-rectangle_8.java
class Solution {
    public boolean isRectangleCover(int[][] rectangles) {
        
        int n = rectangles.length;
        if (n == 0) {
            return true;
        }
        int left = Integer.MAX_VALUE, right = Integer.MIN_VALUE, top = Integer.MIN_VALUE, bottom = Integer.MAX_VALUE;
        Set<String> set = new HashSet<>();
        int area = 0;
        for (int[] rec : rectangles) {
            left = Math.min(left, rec[0]);
            bottom = Math.min(bottom, rec[1]);
            right = Math.max(right, rec[2]);
            top = Math.max(top, rec[3]);
            area += (rec[2] - rec[0]) * (rec[3] - rec[1]);
            String s1 = rec[0] + " " + rec[1];
            String s2 = rec[0] + " " + rec[3];
            String s3 = rec[2] + " " + rec[3];
            String s4 = rec[2] + " " + rec[1];
            if (!set.add(s1)) {
                set.remove(s1);
            }
            if (!set.add(s2)) {
                set.remove(s2);
            }
            if (!set.add(s3)) {
                set.remove(s3);
            }
            if (!set.add(s4)) {
                set.remove(s4);
            }
        }
        if (!set.contains(left + " " + bottom) || !set.contains(left + " " + top) || !set.contains(right + " " + bottom) || !set.contains(right + " " + top) || set.size() != 4) {
            return false;
        }
        return area == (right - left) * (top - bottom);
    }
}